web-ui/src/helpers/query-parameters.js

```diff
@@ -34,13 +47,21 @@ export const useQueryParameters = qps => {
         qp.setter(v || qp.default);
       }
     }
-  }, []);
+    if (processedQPs) processedQPs.current = true;
+  }, requirements);
 
   const dependencies = qps.map(qp => qp.value);
 
+  // This updates the query parameters in the URL
+  // when their associated state values change.
   useEffect(() => {
+    const haveRequirements = requirements.every(req =>
```

Comment:
If the requirements have not been met then we cannot update the query parameters.